### ISSUE TITLE
Logfile definitions

### ISSUE THREAD

**Issue by reinhold-willcox** [TeamCOMPAS/COMPAS#745]
**Describe the bug**
The --logfile_definitions option does not work when the same file REC is supplied twice. I ran with the log_defs.txt file shown below. There were no errors, and the output had correctly removed the zeta parameters but the CO core masses were not included. I've spoken to Jeff about this, and he knows the solution, so I won't elaborate further.

log_defs.txt:
```
BSE_RLOF_REC += { 
    STAR_1_PROPERTY::CO_CORE_MASS,
    STAR_2_PROPERTY::CO_CORE_MASS
}

BSE_RLOF_REC -= { 
    STAR_1_PROPERTY::ZETA_SOBERMAN,
    STAR_1_PROPERTY::ZETA_SOBERMAN_HE,
    STAR_1_PROPERTY::ZETA_HURLEY,
    STAR_1_PROPERTY::ZETA_HURLEY_HE,
    STAR_2_PROPERTY::ZETA_SOBERMAN,
    STAR_2_PROPERTY::ZETA_SOBERMAN_HE,
    STAR_2_PROPERTY::ZETA_HURLEY,
    STAR_2_PROPERTY::ZETA_HURLEY_HE
}
```

**Label the issue**
Please label the 'severity' and 'urgency' of this issue. You can choose:

`urgency_low`      - This issue is not urgent
`severity_minor`    - This is a minor bug with minimal impact

**Versioning (please complete the following information):**
 - COMPAS v02.27.01

**Comment by jeffriley** [comment]
It seems I introduced this when I added annotations.  I remember changing the code in question - I just don't remember exactly why I changed it (I have a vague recollection...).  I thought I tested all combinations of the logfile definitions file, but obviously not.  It shouldn't take long to fix - I'll get to it soon.

**Comment by jeffriley** [comment]
I have pushed a fix for this (PR #757 ), but it comes with a caveat.

I remember changing the offending code (see the PR) when I implemented annotations back in v02.25.00 - I changed the code because annotations and logfile-definitions files weren't working together.  The change I made then caused this defect, and it is obvious why it did that.  I don't remember if I made the change just to test something, or made it and then fixed the underlying problem, and forgot to "un-make" the change.

I've now "un-made" those changes, and I can't break it - but the caveat is that one of my many faults is that I'm not good at thoroughly testing my own code, so it may be that the old underlying problem is still there.  If it is, hopefully somebody will make it break and report the problem (open an issue), and then I'll remember what it was and fix it...
